File: internal/content/testdata/bench/javascript/nc-set-matrix-zeroes.js
function set_zeroes(matrix) {
    const zeroRows = new Set();
    const zeroCols = new Set();
    const rows = matrix.length;
    const cols = matrix[0].length;
    for (let r = 0; r < rows; r++) {
        for (let c = 0; c < cols; c++) {
            if (matrix[r][c] === 0) {
                zeroRows.add(r);
                zeroCols.add(c);
            }
        }
    }
    for (let r = 0; r < rows; r++) {
        for (let c = 0; c < cols; c++) {
            if (zeroRows.has(r) || zeroCols.has(c)) {
                matrix[r][c] = 0;
            }
        }
    }
    return matrix;
}

File: internal/content/testdata/bench/java/nc-set-matrix-zeroes.java
import java.util.*;
class Solution {
    public long[][] set_zeroes(long[][] matrix) {
        Set<Integer> zeroRows = new HashSet<>();
        Set<Integer> zeroCols = new HashSet<>();
        int rows = matrix.length, cols = matrix[0].length;
        for (int r = 0; r < rows; r++)
            for (int c = 0; c < cols; c++)
                if (matrix[r][c] == 0) { zeroRows.add(r); zeroCols.add(c); }
        for (int r = 0; r < rows; r++)
            for (int c = 0; c < cols; c++)
                if (zeroRows.contains(r) || zeroCols.contains(c))
                    matrix[r][c] = 0;
        return matrix;
    }
}
